scripts/scons_helpers/build-3rdparty.py

```diff
@@ -664,18 +664,152 @@ def splitenv(envlist):
         env[k] = v
     return env
 
+# Guess platform argument for OpenSSL's Configure script basing on toolchain string.
+# (see `./Configure LIST` for a list of all platforms and their format)
+# See also openssl/Configurations/[0-9]*.conf
+def openssl_get_platform(toolchain):
+    """
+    >>> openssl_get_platform('x86_64-pc-linux-gnu')
+    'linux-x86_64'
+    >>> openssl_get_platform('aarch64-linux-gnu')
+    'linux-aarch64'
+    >>> openssl_get_platform('mips-openwrt-linux-uclibc')
+    'linux-mips32'
+    >>> openssl_get_platform('arm-linux-gnueabihf')
+    'linux-generic32'
+    >>> openssl_get_platform('arm-bcm2708hardfp-linux-gnueabi')
+    'linux-generic32'
+    >>> openssl_get_platform('mips64-linux-musl')
+    'linux64-mips64'
+    >>> openssl_get_platform('arm-linux-androideabi')
+    'android-arm'
+    >>> openssl_get_platform('aarch64-linux-android')
+    'android-arm64'
+    >>> openssl_get_platform('aarch64-linux-android29')
+    'android-arm64'
+    >>> openssl_get_platform('i686-linux-android')
+    'android-x86'
+    >>> openssl_get_platform('x86_64-linux-android')
+    'android-x86_64'
+    >>> openssl_get_platform('i686-pc-linux-gnu')
+    'linux-x86'
+
+    # other edge cases
+    >>> openssl_get_platform('mingw')
+    'mingw'
+    >>> openssl_get_platform('cygwin')
+    'Cygwin'
```

Comment:
nit: let's replace these inputs in test with real examples:

```
mingw-w64-x86_64
x86_64-pc-cygwin
```

Reply:
done